oneargopy/Argo.py

```diff
@@ -1019,3 +1152,300 @@ def __determine_graph_step(self, ax, axis: str)-> int:
+
+        # Getting the file paths for downloaded .nc files
+        directory = Path(self.download_settings.base_dir.joinpath("Profiles"))
+        file_paths = []
+        for file in files : 
+            file_paths.append(directory.joinpath(file))
+
+        # Columns that will always be in the dataframe, these columns are one dimensional
+        static_columns = ['WMOID', 'CYCLE_NUMBER', 'DIRECTION', 
+                                'DATE', 'DATE_QC', 'LATITUDE', 
+                                'LONGITUDE', 'POSITION_QC',]
+        # Columns that need to be calculated or derived 
+        special_case_static_columns = ['DATE', 'DATE_QC', 'WMOID']
+
+        # Empty Dataframe to return at end of function with all loaded data
+        float_data_dataframe = pd.DataFrame()
+
+        # Iterate through files
+        for file in file_paths :             
+            # Open File
+            nc_file = netCDF4.Dataset(file, mode='r')
+            
+            # Get dimensions of .nc file
+            number_of_profiles = nc_file.dimensions['N_PROF'].size
+            number_of_levels = nc_file.dimensions['N_LEVELS'].size
+
+            # Get float id of current file
+            float_id_array = nc_file.variables['PLATFORM_NUMBER'][0]
+            float_id_array = [elem.decode('utf-8') if isinstance(elem, bytes) else elem for elem in float_id_array]
+            float_id_array.pop() # The last value in the array is 'masked' which we don't need to form the float_id
+            float_id = int(''.join(float_id_array))
+
+            # Get the range of profiles from the index file
+            profile_count = self.prof_index['wmoid'].value_counts().get(float_id, 0)
+
+            # Compare against .nc range to index file range
+            if profile_count > number_of_profiles : 
+                print(f'Skipping float {float_id}...')
+                print(f'The index file has {profile_count} profiles and the .nc file has {number_of_profiles} profiles for float {float_id}..')
+                continue
+
+            # Manage passed profiles if necessary
+            if self.float_profiles_dict is not None : 
+                # Get list of profiles passed in dictionary for float
+                profiles_to_pull = self.float_profiles_dict[float_id]
+
+                # Adjusting profile to index correctly from .nc file arrays
+                profiles_to_pull = [index - 1 for index in profiles_to_pull]
+
+                # The case for if we are pulling a single profile
+                if len(profiles_to_pull) == 1 : 
+                    profiles_to_pull = int(profiles_to_pull[0])
+                    static_length = 1
+                else : 
+                    static_length = len(profiles_to_pull)
+            # If no profiles are passed then we want to pull all of the profiles from the float
+            else : 
+                profiles_to_pull = list(range(0, number_of_profiles-1, 1))
```

Comment:
we want all the profiles, why are you subtracting 1 (here and next line)??

Reply:
ah i think i was having an indexing issue at some point and this is a remnant of that. you're right these subtractions shouldn't be there.